frontend/src/components/CognitoLoginAndReg.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { useEffect, useState } from 'react';
+import Button from 'react-bootstrap/Button';
+import { useLocation } from 'react-router-dom'
+
+function useQuery() {
+  return new URLSearchParams(useLocation().search);
+}
+
+export default () => {
+  const [, setTokenObj ] = useState({});
+  const [, setUserObj ] = useState({});
+
+  let query = useQuery();
+  let code = query.get("code");
+
+  function fetchTokens() {
```

Comment:
Could you please move the API functions to a separate file? Maybe something like `src/apis/cognito.jsx`? It will separate the API layer and be consistent with `src/apis/infomed.tsx`.